src/dotnet/ExecutablePackageObtainer/PackageVersion.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.DotNet.ExecutablePackageObtainer
+{
+    internal class PackageVersion
+    {
+        public PackageVersion(string packageVersion)
+        {
+            if (packageVersion == null)
+            {
+                Value = Path.GetRandomFileName();
+                IsPlaceHolder = true;
+            }
+            else
+            {
+                Value = packageVersion;
+                IsPlaceHolder = false;
+            }
+        }
+
+        public bool IsPlaceHolder { get; }
```

Comment:
Would it make sense to remove this property in favor of just using `!IsConcreteValue` given that is just the inverse of `IsPlaceHolder`?

Reply:
no strong opinion, this is a habit learnt from ruby. This make the if statement a little easier to read. It can help when there is a lot of bit flipping.

Is there big concern here? Or i'd like to keep it

Reply:
No big concern; it was just initially confusing as to what constituted a check for being a "placeholder" vs a "not concrete" version.

Reply:
Minor thing here: placeholder is one word not, so it should be spelled IsPlaceholder.

Reply:
Fixed